test/unit/org/apache/cassandra/db/SchemaCQLHelperTest.java

```diff
@@ -444,13 +446,17 @@ public void testSnapshot() throws Throwable
                          containsString("ALTER TABLE " + keyspace() + "." + tableName + " DROP reg3 USING TIMESTAMP 10000;"),
                          containsString("ALTER TABLE " + keyspace() + "." + tableName + " ADD reg3 int;")));
 
-        assertThat(schema, containsString("CREATE INDEX IF NOT EXISTS " + tableName + "_reg2_idx ON " + keyspace() + '.' + tableName + " (reg2);"));
+        final boolean isIndexLegacy = DatabaseDescriptor.getDefaultSecondaryIndex().equals(CassandraIndex.NAME);
```

Comment:
I wonder if it would read better to move this that is repeated so much into a method in CQLTester with some javadoc explanation

Reply:
Created `Util.assumeLegacySecondaryIndex()` to replace most of the repetition.

It doesn't change this particular line, though. Let me know if you insist on that being placed in `Util` as well.

Reply:
sgtm